src/main/java/org/opentripplanner/middleware/controllers/api/UserController.java

```diff
@@ -67,4 +95,44 @@ boolean preDeleteHook(User user, Request req) {
         }
         return true;
     }
+
+    /**
+     * Holds result and message from getUserFromProfile.
+     */
+    static class UserFromProfileResult {
+        public User user;
+        public String message;
+    }
+
+    /**
+     * HTTP endpoint to get the {@link User} entity from a {@link Auth0UserProfile}.
+     * (Reminder: for endpoints under 'secure', we add a {@link Auth0UserProfile} to request attributes.)
+     */
+    private User getUserFromRequest(Request req, Response res) {
+        Auth0UserProfile profile = req.attribute("user");
+        UserFromProfileResult result = getUserFromProfile(profile);
+
+        if (result.user == null) {
+            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, result.message,null);
+        }
+        return result.user;
+    }
+
+    /**
+     * @param profile The {@link Auth0UserProfile} from which to extract the User.
+     * @return An object containing the {@link User} entity from a {@link Auth0UserProfile}, or null and and an error message if that fails.
+     */
+    UserFromProfileResult getUserFromProfile(Auth0UserProfile profile) {
+        UserFromProfileResult result = new UserFromProfileResult();
+
+        if (profile != null) {
+            String auth0UserId = profile.user_id;
+            result.user = persistence.getOneFiltered(eq("auth0UserId", auth0UserId));
+            if (result.user == null) result.message = String.format(NO_USER_WITH_AUTH0_ID_MESSAGE, auth0UserId);
+        } else {
+            result.message = NO_AUTH0_PROFILE_MESSAGE;
+        }
```

Comment:
This else condition should never happen on an endpoint in this controller (see https://github.com/ibi-group/otp-middleware/pull/22/files#r432599004).

Reply:
Hmm good point. We should consider making that profile a field?